models/room.go

```diff
@@ -603,3 +603,92 @@ func GetRoomsByMetric(redisClient interfaces.RedisClient, schedulerName string,
 		}
 	}
 }
+
+// GetInvalidRoomsKey gets the key for the set that keeps names of invalid rooms (rooms from old scheduler version)
+func GetInvalidRoomsKey(schedulerName string) string {
+	return fmt.Sprintf("scheduler:%s:invalidRooms", schedulerName)
+}
+
+// GetInvalidRoomsCountKey gets the key for the string that keeps count of invalid rooms
+func GetInvalidRoomsCountKey(schedulerName string) string {
+	return fmt.Sprintf("scheduler:%s:invalidRooms:count", schedulerName)
+}
+
+// SetInvalidRooms save a room in invalid redis set
+// A room is considered invalid if its version is not the scheduler current version
+func SetInvalidRooms(redisClient interfaces.RedisClient, mr *MixedMetricsReporter, schedulerName string, roomIDs []string) error {
+	pipe := redisClient.TxPipeline()
+	pipe.Del(GetInvalidRoomsKey(schedulerName))
+	pipe.Del(GetInvalidRoomsCountKey(schedulerName))
+	pipe.SAdd(GetInvalidRoomsKey(schedulerName), roomIDs)
+	pipe.Set(GetInvalidRoomsCountKey(schedulerName), len(roomIDs), 2*time.Hour)
+	err := mr.WithSegment(SegmentPipeExec, func() error {
+		var err error
+		_, err = pipe.Exec()
+		return err
+	})
+	return err
+}
+
+// RemoveInvalidRooms deletes an invalid room
+func RemoveInvalidRooms(redisClient interfaces.RedisClient, mr *MixedMetricsReporter, schedulerName string, roomIDs []string) error {
+	pipe := redisClient.TxPipeline()
+	pipe.SRem(GetInvalidRoomsKey(schedulerName), roomIDs)
+	err := mr.WithSegment(SegmentPipeExec, func() error {
+		var err error
+		_, err = pipe.Exec()
+		if err == redis.Nil {
+			return nil
+		}
+		return err
+	})
+	return err
+}
+
+// RemoveInvalidRoomsKey deletes invalidRooms redis key
+func RemoveInvalidRoomsKey(redisClient interfaces.RedisClient, mr *MixedMetricsReporter, schedulerName string) error {
+	pipe := redisClient.TxPipeline()
+	pipe.Del(GetInvalidRoomsKey(schedulerName))
+	pipe.Del(GetInvalidRoomsCountKey(schedulerName))
+	err := mr.WithSegment(SegmentPipeExec, func() error {
+		var err error
+		_, err = pipe.Exec()
+		return err
+	})
+	return err
+}
+
+// GetCurrentInvalidRoomsCount returns the count of current invalid rooms
+func GetCurrentInvalidRoomsCount(redisClient interfaces.RedisClient, mr *MixedMetricsReporter, schedulerName string) (int, error) {
+	count := 0
+	err := mr.WithSegment(SegmentSMembers, func() error {
+		var err error
+		keys, err := redisClient.SMembers(GetInvalidRoomsKey(schedulerName)).Result()
```

Comment:
It's better to use a lua script that returns the count than loading all members in memory and counting

Reply:
actually, using SCARD (https://redis.io/commands/scard) would be the best choice